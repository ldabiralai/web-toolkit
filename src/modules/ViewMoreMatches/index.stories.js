import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import styled from 'react-emotion';
import { object } from '@storybook/addon-knobs';
import { ViewMoreMatches } from '../..';
import { coreLightMinus1 } from '../../colors';

const viewmorematches = storiesOf('Modules|ViewMoreMatches', module).addDecorator(withInfo);

const Wrapper = styled.div`
  color: ${coreLightMinus1};
`;

const Element = styled.div`
  margin-bottom: 15px;
`;

const children = [];
for (let i = 0; i < 10; i += 1) {
  children.push(<Element key={i}>match</Element>);
}

viewmorematches.add('viewmorematches', () => (
  <Wrapper>
    <ViewMoreMatches>{object('children', children)}</ViewMoreMatches>
  </Wrapper>
));
