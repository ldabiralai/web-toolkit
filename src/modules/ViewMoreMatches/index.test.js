import React from 'react';
import { mount } from 'enzyme';
import { object } from '@storybook/addon-knobs';
import ViewMoreMatches from '.';

const children = [];
for (let i = 0; i < 10; i += 1) {
  children.push(<div key={i}>match</div>);
}

describe('ViewMoreMatches', () => {
  it('renders its content', () => {
    expect(mount(<ViewMoreMatches>{object('children', children)}</ViewMoreMatches>)).toMatchSnapshot();
  });
});
