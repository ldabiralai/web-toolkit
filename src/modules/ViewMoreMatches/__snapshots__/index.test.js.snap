// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewMoreMatches renders its content 1`] = `
<ViewMoreMatches>
  <Styled(ul)
    blockHeight={0}
    innerRef={
      Object {
        "current": <ul
          class="css-12eg17e e1gwa8591"
        >
          <li
            class="css-1u41msn e1gwa8592"
          >
            <div>
              match
            </div>
          </li>
        </ul>,
      }
    }
    isExpanded={false}
  >
    <ul
      className="css-12eg17e e1gwa8591"
    >
      <Styled(li)
        key="0"
      >
        <li
          className="css-1u41msn e1gwa8592"
        >
          <div
            key="0"
          >
            match
          </div>
        </li>
      </Styled(li)>
    </ul>
  </Styled(ul)>
  <Styled(span)
    onClick={[Function]}
  >
    <span
      className="css-41z9q8 e1gwa8590"
      onClick={[Function]}
    >
      Show more
    </span>
  </Styled(span)>
</ViewMoreMatches>
`;
